test/basics.spec.ts

```diff
@@ -58,15 +58,16 @@ describe('basics', () => {
     remoteNode = await createNode()
   })
 
-  afterEach(async () => {
-    if (localNode != null) {
-      await localNode.stop()
-    }
+  // This is currently causing issues on macOS runs
```

Comment:
There seems to be a non-deterministic bug during the teardown of our relay tests which results in an error being thrown on webkit tests, specifically when running on my local (macOS) where the `RTCDataChannel` is unable to [send the encoded `STOP_SENDING` flag](https://github.com/libp2p/js-libp2p-webrtc/blob/test/run-tests-on-webkit/src/stream.ts#L190).  

``` bash
Error sending binary data through RTCDataChannel.
```
See image for more details.
![image](https://github.com/libp2p/js-libp2p-webrtc/assets/9755286/7a5d5295-ccf2-43d2-b4e0-2396dcca47d5)

Reply:
Are you sure we're setting the binaryType correctly everytime we create a datachannel? I think we might be missing these:

```
diff --git a/src/private-to-private/handler.ts b/src/private-to-private/handler.ts
index 176031b..793477e 100644
--- a/src/private-to-private/handler.ts
+++ b/src/private-to-private/handler.ts
@@ -104,6 +104,7 @@ export async function initiateConnection ({ rtcConfiguration, dataChannelOptions
   // to collect candidates. The label is not relevant to connection initiation
   // but can be useful for debugging
   const channel = pc.createDataChannel('init')
+  channel.binaryType = 'arraybuffer'
   // setup callback to write ICE candidates to the remote
   // peer
   pc.onicecandidate = ({ candidate }) => {
diff --git a/src/private-to-public/transport.ts b/src/private-to-public/transport.ts
index c23b715..ac4fdd7 100644
--- a/src/private-to-public/transport.ts
+++ b/src/private-to-public/transport.ts
@@ -139,6 +139,7 @@ export class WebRTCDirectTransport implements Transport {
     // the peer.
     const dataChannelOpenPromise = new Promise<RTCDataChannel>((resolve, reject) => {
       const handshakeDataChannel = peerConnection.createDataChannel('', { negotiated: true, id: 0 })
+      handshakeDataChannel.binaryType = 'arraybuffer'
       const handshakeTimeout = setTimeout(() => {
         const error = `Data channel was never opened: state: ${handshakeDataChannel.readyState}`
         log.error(error)

```

Reply:
I think so as [all streams are initialized as such](https://github.com/libp2p/js-libp2p-webrtc/blob/test/run-tests-on-webkit/src/stream.ts#L66) and even if it was set to `blob` ([the default](https://developer.mozilla.org/en-US/docs/Web/API/RTCDataChannel/binaryType)) then we would have seen errors being thrown in [other tests](https://github.com/libp2p/js-libp2p-webrtc/blob/test/run-tests-on-webkit/test/stream.browser.spec.ts#L12) but I did a sanity check and added those changes as well and the result was the same. As I had mentioned the errors are thrown **after** the teardown of the tests - i.e. the test actually passes.

Reply:
I do notice that also [the setup was moved on this PR as well](https://github.com/libp2p/js-libp2p-webrtc/pull/175/files#diff-82800680dd98113b2d5f557f41fbfca2a686b7dc84fe106e432d86367a8d15a5R88-R96) I am not sure if that has any relation.